libp2p/connmanager.nim

```diff
@@ -105,15 +109,34 @@ proc addConnEventHandler*(c: ConnManager,
   ## Add peer event handler - handlers must not raise exceptions!
   ##
 
-  if isNil(handler): return
-  c.connEvents.mgetOrPut(kind,
-    initOrderedSet[ConnEventHandler]()).incl(handler)
+  try:
+    if isNil(handler): return
+    c.connEvents.mgetOrPut(kind,
+      initOrderedSet[ConnEventHandler]()).incl(handler)
+  except Exception as exc:
+    # TODO: there is an Exception being raised
```

Comment:
this is probably https://github.com/nim-lang/Nim/issues/17382 - looks like a nim bug, can use it in the comment to highlight where it happens

Reply:
also, it incorrectly raises whatever `handler` raises - since you see `Exception` here it means `handler` itself is missing a correct raises signature

Reply:
`ConnEventHandler` is defined as:

```nim
  ConnEventHandler* =
    proc(peerId: PeerID, event: ConnEvent): Future[void]
      {.gcsafe, raises: [Defect].}
```

But interestingly, it seems to ignore that when actually declaring the handler which is declared in https://github.com/status-im/nim-libp2p/blob/unstable/tests/testswitch.nim#L249. Another bug?